docker rootless info (#2274)
alan-turing-institute/the-turing-way: book/website/reproducible-research/renv/renv-containers.md

There are many tools available for creating and working with containers.
We will focus on Docker, which is widely used, but be aware that others such as Singularity also exist.

> Singularity is sometimes preferred for use on high-performance computing systems as it does not need `sudo` permissions to be run, while up until April 2020 Docker did (please see the [Docker without root access](#docker-without-root-access) section)

Reviewer: ```suggestion
Singularity is sometimes preferred for use on high-performance computing systems as it does not need `sudo` permissions to be run, while up until April 2020 Docker did (please see the {ref}`rr-renv-containers-rootless` section).
```

Author: 👍

When they come back to the project and want to continue their work, they only need to use the same command as above, and it will load the work contained in `volume_name` into the new container.
It will save any new work there too.

Below is a list of volume related commands: 

- To list volumes: `sudo docker volume ls`
- To delete a volume: `sudo docker volume rm volume_name`
- To delete all unattached volumes: `sudo docker volume prune`

If, when deleting a container, a `-v` is included after `rm` in `sudo docker rm container_ID`, any volumes associated with the container will also be deleted.

(rr-renv-containers-rootless)=
## Docker without root access

Up until April 2020, the only way to run Docker was with root access. 
"Rootless" mode was made available as part of the [v20.10](https://docs.docker.com/engine/security/rootless/) release.
 Rootless mode is currently only avaliable on Linux and requires an initial install of Docker >= v20.10. 
 

>  The underyling difference between Docker and rootless mode is that perviously any system running docker had a daemon running as `uid0` that creates and owns all images, but with rootless mode the user creates and owns any images that they initialize.

Reviewer: ```suggestion
 The underyling difference between Docker without and with rootless mode is that perviously any system running Docker had a daemon running as `uid0` that creates and owns all images, but with rootless mode the user creates and owns any images that they initialize.
```

Author: 👍

 To install and run the rootless version of Docker as a non-root user, use the following commands:

```
dockerd-rootless-setuptool.sh install

> docker run -d --name dind-rootless --privileged docker:20.10-dind-rootless

Reviewer: Does `20.10` refer to the installed version of Docker?

Author: that's correct!

```

> The following prequisites are required to run Docker rootless: `newuidmap` and `newgidmap`. 

Reviewer: ```suggestion
The following prequisites, which are part of the [`shadow-utils`](https://github.com/shadow-maint/shadow) package are required to run Docker rootless: `newuidmap` and `newgidmap`. 
```

Author: 👍
